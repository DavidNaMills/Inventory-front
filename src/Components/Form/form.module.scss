@import '../../styles/variables/sizing.module.scss';
@import '../../styles/variables/colors.module.scss';

@mixin form {
    -webkit-backdrop-filter: blur(5px);
    backdrop-filter: blur(5px);
    // background-color: rgba(242, 242, 242, 0.5);

    margin-bottom: $largeSpacing;
    margin-top: $mediumSpacing;
    z-index: 1;
}



.form{
    border-radius: 3px;
    padding: 15px;
    
    
    -webkit-box-shadow: $blur $shadow-blue;
    -moz-box-shadow: $blur $shadow-blue;
    box-shadow: $blur $shadow-blue;

    &__buttonSpacing{
        &_large{
            margin-top: $mediumSpacing;
        }
        &_medium{
            margin-top: $extraSmallSpacing;
        }
    }
}

.blur{
    &__normal{
        width: 40%;
        @include form;
    }
    
    &_mobile{
        width: 80%;
        @include form;
    }
}




// @import '../../styles/variables/sizing.module.scss';
// @import '../../styles/variables/colors.module.scss';

// @mixin form {
//     border-radius: 3px;
//     padding: 15px;
//     margin-bottom: $largeSpacing;
//     margin-top: $mediumSpacing;
    
//     -webkit-box-shadow: $blur $shadow-blue;
//     -moz-box-shadow: $blur $shadow-blue;
//     box-shadow: $blur $shadow-blue;


// }


// .form{
//     &__normal{
//         width: 40%;
//         @include form;
//     }
    
//     &_mobile{
//         width: 80%;
//         @include form;
//     }

//     &__buttonSpacing{
//         &_large{
//             margin-top: $mediumSpacing;
//         }
//         &_medium{
//             margin-top: $extraSmallSpacing;
//         }
//     }
// }

// .blur{
//     // width:100%;
//     -webkit-backdrop-filter: blur(10px);
//     backdrop-filter: blur(5px);
//     background-color: rgba(255, 255, 255, 0.5); 
// }