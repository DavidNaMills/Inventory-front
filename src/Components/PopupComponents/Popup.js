import React from 'react';
import classes from './Popup.module.scss';
import usePositionHook from '../../hooks/usePositionHook/usePositionHook';

const style = (pos) => ({
    position: 'absolute',
    top: `${pos}px`,
    left: '0px',
    width: '100%',
    height: '100vh',
    backgroundColor: 'rgba(232, 255, 253, 0.6)',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    zIndex: '9999'
})

const Popup = (props) => {
    const { children, isOpen = false, onClose = null } = props;
    const { posRef } = usePositionHook();
    
    return (
        isOpen ?

            <div style={style(posRef)}>
                <div>
                    {children}
                </div>
            </div>
            : null
    )
}

export default Popup;