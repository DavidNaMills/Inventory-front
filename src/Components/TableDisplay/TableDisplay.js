import React from 'react';
import classes from './tableDisplay.module.scss';
import { useLocale } from '../../Context/LocaleContext';
import { isMobileOnly } from 'react-device-detect';
import StaffTypeRow from './Row/StaffTypeRow';
import ProductTypeRow from './Row/ProductTypeRow/ProductTypeRow';
import ProductRow from './Row/ProductRow/ProductRow';
import Button from '../Button/Button';
import MoreHorizIcon from '@material-ui/icons/MoreHoriz';
import CreateIcon from '@material-ui/icons/Create';


export const TableDisplay = (props) => {
    const locale = useLocale();
    const { list, config, update = null, remove = null, formName, viewMore = null, isSelected=null } = props;

    const all = list.map((x, count) => {
        let temp = [];
        const tStyle = [classes.tableDisplay];

        if (isSelected === x._id) { tStyle.push(classes.tableDisplay__selected) }
        else if (count % 2) { tStyle.push(classes.tableDisplay__even) }
        else { tStyle.push(classes.tableDisplay__odd) }

        switch (formName) {
            case 'staff':
                temp.push(
                    <StaffTypeRow
                        key={count}
                        config={config}
                        data={x}
                        showInfo={isMobileOnly ? false : true}
                    />
                )
                break;
            case 'productTypes':
                temp.push(
                    <ProductTypeRow
                        key={count}
                        config={config}
                        data={x}
                    />
                )
                break;

            case 'locationTypes':
                temp.push(
                    <ProductTypeRow
                        key={count}
                        config={config}
                        data={x}
                    />
                )
                break;
            case 'products':
                temp.push(
                    <ProductRow
                        key={count}
                        config={config}
                        data={x}
                        withImage={isMobileOnly? false : true}
                        withInfo={isMobileOnly? false : true}
                    />
                )
                break;
            default:
                return <p key={count}>ERROR</p>
        }
        return (
            <div className={tStyle.join(' ')} key={count}>
                {temp}
                <div className={classes.tableDisplay__optionsStyle}>
                    {update && temp.length > 0 &&
                        <Button onClick={() => update(JSON.parse(JSON.stringify(x)))} noStyle Icon={CreateIcon} />
                    }
                    {remove && temp.length > 0 && <Button onClick={() => remove(x._id)} type='danger' label={locale.remove} />}
                    {viewMore && <Button onClick={() => viewMore(x)} type='base' noStyle Icon={MoreHorizIcon} />}
                </div>
            </div>
        )
    })

    return (
        <div>
            { 
                all.length > 0
                ? all
                : <div className={classes.tableDisplay__message}><p className={classes.tableDisplay__message_text}>Nothing to display</p></div>
            }
            
        </div>
    );
}

export default TableDisplay;