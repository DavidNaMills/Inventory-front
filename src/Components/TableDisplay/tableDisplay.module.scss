@import '../../styles/variables/colors.module.scss';

.tableDisplay{
    display: flex;
    justify-content: space-between;
    padding: 5px;

    &__odd{
        background-color: $odd-back;
        border-bottom: solid $odd-border 2px;
    }

    &__even{
        background-color: $even-back;
        border-bottom: solid $even-border 2px;
    }

    &__selected{
        background-color: $selected-back;
    }

    &__optionsStyle {
        display: flex;
        justify-content: space-between;
        align-items: center;
        width: 150px;

        // margin-left: 15px;
        // margin-right: 15px
    }

    &__message{
        width: 100%;
        display: flex;
        justify-content: center;

        &_text{
            font-size: 30px;
            color: $title-color;
        }
    }
}