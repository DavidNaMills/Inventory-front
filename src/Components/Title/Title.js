import React from 'react';
import classes from './title.module.scss';


const Title = (props) => {
    const { size = null, message = '', center = false } = props;
    let element = null;

    switch (size) {
        case 'h1':
            element = <h1 className={[classes.title_h1, center ? classes.title__center : null].join(' ')} >{message}</h1>
            break;
        case 'h2':
            element = <h2 className={[classes.title_h2, center ? classes.title__center : null].join(' ')}>{message}</h2>
            break;
        case 'h3':
            element = <h3 className={[classes.title_h3, center ? classes.title__center : null].join(' ')}>{message}</h3>
            break;
        case 'h4':
            element = <h4 className={[classes.title_h4, center ? classes.title__center : null].join(' ')}>{message}</h4>
            break;
        case 'h5':
            element = <h5 className={[classes.title_h5, center ? classes.title__center : null].join(' ')}>{message}</h5>
            break;
        case 'h6':
            element = <h6 className={[classes.title_h6, center ? classes.title__center : null].join(' ')}>{message}</h6>
            break;
        default:
            element = <h2 className={[classes.title_h2, center ? classes.title__center : null].join(' ')}>{message}</h2>
            break;
    };

    return (
        <div>
            {element}
        </div>
    )
}

export default React.memo(Title);