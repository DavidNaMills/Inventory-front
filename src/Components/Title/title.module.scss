@import '../../styles/variables/colors.module.scss';
@import '../../styles/fonts/fontSizes.module.scss';
@import '../../styles/mixins/title.module.scss';

.title {
    @include setTitle($h2, $title-color);
    
    &_h1{
        @include setTitle($h1, $title-color);
    };

    &_h2{
        @include setTitle($h2, $title-color);
    };
    
    &_h3{
        @include setTitle($h3, $title-color);
    }

    &_h4{
        @include setTitle($h4, $title-color);
    }

    &_h5{
        @include setTitle($h5, $title-color);
    }

    &_h6{
        @include setTitle($h6, $title-color);
    }

    &__center{
        text-align: center;
    }


}