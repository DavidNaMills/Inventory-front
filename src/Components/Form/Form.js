import React from 'react';
import classes from './form.module.scss';
import { isMobileOnly } from 'react-device-detect';
import { useLocale } from '../../Context/LocaleContext';
import FormFactory from './Factory/FormFactory';
import Button from '../Button/Button';
import Title from '../Title/Title';
import PublishIcon from '@material-ui/icons/Publish';
import ClearIcon from '@material-ui/icons/Clear';

import objToArray from '../../helpers/objToArray/objToArray';


const Form = (props) => {
    const locale = useLocale();
    const {
        displayFields,
        user,
        fullForm,
        controlForm,
        updating,
        withTitle = true,

        submitCb = null,
        clearCb = null
    } = props;
    // isMobileOnly ? classes.form_mobile : classes.form__normal}
    return (
        <div className={isMobileOnly ? classes.blur_mobile : classes.blur__normal}>
            <div className={classes.form}>
                {withTitle && <Title message={updating ? `${locale.updating} ${updating}` : locale.creating} size='h3' />}
                <form onSubmit={submitCb}>
                    {
                        objToArray(fullForm).map((x, y) =>
                            <FormFactory
                                key={y}
                                config={x}
                                role={user.role}
                                displayFields={displayFields}
                                onChange={controlForm}
                            />
                        )
                    }
                    <div className={classes.form__buttonSpacing_large}>
                        <Button config={{ type: 'submit' }} type="submit" label={updating ? locale.update : locale.submit} Icon={PublishIcon} full />
                    </div>
                </form>
                {clearCb && <div className={classes.form__buttonSpacing_medium}>
                    <Button onClick={clearCb} label={locale.clear} type='clear' Icon={ClearIcon} full />
                </div>}
            </div>
        </div>
    )
}

export default Form;