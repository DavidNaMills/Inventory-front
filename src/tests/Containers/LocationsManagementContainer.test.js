import React from 'react';
import '../testHelpers/context';
import './containerHelper';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';

import LocationManagementContainer from '../../Containers/LocationsManagementContainer/LocationsManagementContainer';
import Form from '../../Components/Form/Form';
import TableDisplay from '../../Components/TableDisplay/TableDisplay';
import Popup from '../../Components/PopupComponents/Popup';
import Title from '../../Components/Title/Title';




describe('<LocationManagementContainer/> test suite', ()=>{
    let wrapper;

    beforeEach(()=>{
        wrapper = shallow(<LocationManagementContainer />);
    })

    it('matches snapshot', ()=>{
        const comp = renderer.create(<LocationManagementContainer />);
        const app = comp.toJSON();
        expect(app).toMatchSnapshot();
    });
    
    it('exists', ()=>{
        expect(wrapper.exists).toBeTruthy();
    });
    
    it('renders a <Title/> component', ()=>{
        const title = wrapper.find(Title);
        expect(title.length).toBe(1);
        expect(title.props()).toHaveProperty('size', 'h1');
        expect(title.props()).toHaveProperty('message', 'Locations Management');
        expect(title.props()).toHaveProperty('center', true);
    });
    
    it('renders a <Form/> component', ()=>{
        const form = wrapper.find(Form);
        expect(form.length).toBe(1);
        expect(form.props()).toHaveProperty('formName', 'locationTypes');
    });
    
    it('renders a <TableDisplay/> component', ()=>{
        const td = wrapper.find(TableDisplay);
        expect(td.length).toBe(1);
        expect(td.props()).toHaveProperty('formName', 'locationTypes');
        expect(td.props()).toHaveProperty('isSelected', null);

    });
    
    it('renders a <Popup/> component', ()=>{
        expect(wrapper.find(Popup).length).toBe(1);
    });
});