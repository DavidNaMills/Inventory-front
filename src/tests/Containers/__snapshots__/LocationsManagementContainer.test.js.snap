// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LocationManagementContainer/> test suite matches snapshot 1`] = `
<div>
  <div>
    <h1
      className="title_h1 title__center"
    >
      Locations Management
    </h1>
  </div>
  <div
    className="container"
  >
    <div
      className="blur__normal"
    >
      <div
        className="form"
      >
        <div>
          <h3
            className="title_h3 "
          >
            Create new record
          </h3>
        </div>
        <form
          onSubmit={[Function]}
        >
          <div
            className="form__buttonSpacing_large"
          >
            <button
              className="button button__submit button_full"
              disabled={false}
              style={Object {}}
              type="submit"
            >
              <p>
                Submit
              </p>
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M5 4v2h14V4H5zm0 10h4v6h6v-6h4l-7-7-7 7z"
                />
              </svg>
            </button>
          </div>
        </form>
        <div
          className="form__buttonSpacing_medium"
        >
          <button
            className="button button__clear button_full"
            disabled={false}
            onClick={[Function]}
            style={Object {}}
          >
            <p>
              Clear
            </p>
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="tableSpacing"
  >
    <div>
      <div
        className="tableDisplay tableDisplay__odd"
      >
        <div
          className="productTypeRow"
        >
          <div
            className="productTypeRow__title"
          >
            <div>
              <h4
                className="title_h4 "
              >
                home
              </h4>
            </div>
          </div>
          <div
            className="productTypeRow__cellCont"
          >
            <p>
              <span>
                Description
                :
              </span>
            </p>
          </div>
        </div>
        <div
          className="tableDisplay__optionsStyle"
        >
          <button
            className="button button_noStyle"
            disabled={false}
            onClick={[Function]}
            style={Object {}}
          >
            <p />
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
              />
            </svg>
          </button>
          <button
            className="button button_noStyle"
            disabled={false}
            onClick={[Function]}
            style={Object {}}
          >
            <p />
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        className="tableDisplay tableDisplay__even"
      >
        <div
          className="productTypeRow"
        >
          <div
            className="productTypeRow__title"
          >
            <div>
              <h4
                className="title_h4 "
              >
                home2
              </h4>
            </div>
          </div>
          <div
            className="productTypeRow__cellCont"
          >
            <p>
              <span>
                Description
                :
              </span>
            </p>
          </div>
        </div>
        <div
          className="tableDisplay__optionsStyle"
        >
          <button
            className="button button_noStyle"
            disabled={false}
            onClick={[Function]}
            style={Object {}}
          >
            <p />
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
              />
            </svg>
          </button>
          <button
            className="button button_noStyle"
            disabled={false}
            onClick={[Function]}
            style={Object {}}
          >
            <p />
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
